From 78b79d9a98c46db3eef3e1cbca45f7b92a86e41d Mon Sep 17 00:00:00 2001
From: Vasyl Vavrychuk <vasyl.vavrychuk@opensynergy.com>
Date: Mon, 25 Apr 2022 10:59:44 +0300
Subject: [PATCH] Remove 'psplash-fb.h' from 'psplash.h'

psplash might not be necessary based on framebuffer, it could use DRM
in future too.

Signed-off-by: Vasyl Vavrychuk <vasyl.vavrychuk@opensynergy.com>
---
 psplash-fb.c | 2 +-
 psplash-fb.h | 3 +++
 psplash.c    | 1 +
 psplash.h    | 2 --
 4 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/psplash-fb.c b/psplash-fb.c
index 1d2d7db..5dea82a 100644
--- a/psplash-fb.c
+++ b/psplash-fb.c
@@ -8,7 +8,7 @@
  */
 
 #include <endian.h>
-#include "psplash.h"
+#include "psplash-fb.h"
 
 static void
 psplash_wait_for_vsync(PSplashFB *fb)
diff --git a/psplash-fb.h b/psplash-fb.h
index 6c4599c..4d5c460 100644
--- a/psplash-fb.h
+++ b/psplash-fb.h
@@ -10,6 +10,9 @@
 #ifndef _HAVE_PSPLASH_FB_H
 #define _HAVE_PSPLASH_FB_H
 
+#include <linux/fb.h>
+#include "psplash.h"
+
 enum RGBMode {
     RGB565,
     BGR565,
diff --git a/psplash.c b/psplash.c
index 62244ba..18c012b 100644
--- a/psplash.c
+++ b/psplash.c
@@ -11,6 +11,7 @@
  */
 
 #include "psplash.h"
+#include "psplash-fb.h"
 #include "psplash-config.h"
 #include "psplash-colors.h"
 #include "psplash-poky-img.h"
diff --git a/psplash.h b/psplash.h
index 8e68b7b..7489e75 100644
--- a/psplash.h
+++ b/psplash.h
@@ -15,7 +15,6 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <limits.h>
-#include <linux/fb.h>
 #include <linux/kd.h>
 #include <linux/vt.h>
 #include <signal.h>
@@ -78,7 +77,6 @@ typedef struct PSplashFont
 PSplashFont;
 
 
-#include "psplash-fb.h"
 #include "psplash-console.h"
 
 #endif
